test/events/test_maintain_orcid_webhook.py

```diff
@@ -90,6 +96,49 @@ def test_it_ignores_other_models_being_committed(orcid_token: OrcidToken,
     assert mock_orcid_client.remove_webhook.call_count == 0
 
 
-def test_it_has_a_valid_signal_handler_registered_on_app():
-    registered_handler_names = [recv.__name__ for id_, recv in models_committed.receivers.items()]
-    assert 'webhook_maintainer' in registered_handler_names
+@patch('profiles.events.catch_exceptions')
+def test_exception_not_handled_if_catch_decorator_is_removed(profile: Profile,
+                                                             orcid_config: Dict[str, str],
+                                                             mock_orcid_client: MagicMock,
+                                                             session: scoped_session,
+                                                             url_safe_serializer: URLSafeSerializer
+                                                             ):
+    with pytest.raises(Exception):
```

Comment:
This shouldn't cover the whole test? Removing it sees an earlier exception raised.

Reply:
With `mock_catch` removed `session.add(profile)` will throw an error because the `@patch` decorator needs to assign itself to something, and it will be the first arg (`profile` with it removed). So it was trying to add a MagicMock to the session. I will move the exception detection further down, the role of the test was to show it handles unknown exceptions so it did work but granted it should either detect the last part of the test.

Reply:
Oh I see; I get the same error at a different at a later point with it present (after removing `with pytest.raises(Exception)`).

Reply:
I am removing the two tests that basically remove the catch_exceptions decorator and test for failures. I have wrestled with mock based errors and they are not working as needed in this scenario. We have the tests for the actual decorator and we also have a test now for each handler raising and handling exceptions, I think it covers it.